Mathlib/GroupTheory/FiniteAbelian.lean

```diff
@@ -14,14 +14,73 @@ import Mathlib.Data.ZMod.Quotient
 * `AddCommGroup.equiv_free_prod_directSum_zmod` : Any finitely generated abelian group is the
   product of a power of `ℤ` and a direct sum of some `ZMod (p i ^ e i)` for some prime powers
   `p i ^ e i`.
-* `AddCommGroup.equiv_directSum_zmod_of_fintype` : Any finite abelian group is a direct sum of
+* `AddCommGroup.equiv_directSum_zmod_of_finite` : Any finite abelian group is a direct sum of
   some `ZMod (p i ^ e i)` for some prime powers `p i ^ e i`.
 
 -/
 
-
 open scoped DirectSum
 
+/-
+TODO: Here's a more general approach to dropping trivial factors from a direct sums:
```

Comment:
Please indicate briefly what it takes to fill in the sorries. Just hard work, or are there prerequisites missing?

Reply:
Just hard work probably